### PR TITLE
Fix: blue screen issue on Safari Web

### LINKED ISSUE #2705: mWeb -Chat - Weird behavior with 3D Touch on chat screen
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
A context menu should open without issues

## Actual Result:
Context menu opens with blue flashing or opens with overlapping white background on mWeb iOS

## Action Performed:
1. Navigate to https://staging.expensify.cash/
2. Login with existing account
3. Open any chat
4. Long press on any message

## Workaround:
Unknown 

## Platform:
Where is this issue occurring?

 Web
 iOS
 Android
 Desktop App
 Mobile Web ✔️ iOS only on Safari

**Version Number:** 1.0.38-0
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:**  Issue occurs inconsistently (3/5) in iOS (iPhone 12 Pro - iOS 14.4)
**Expensify/Expensify Issue URL:**

https://user-images.githubusercontent.com/43996225/117212784-c7146880-adc8-11eb-88a9-e3a156d2af9c.mp4

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ Fixes #2705

### Tests | QA Steps
1. Open Any conversation on E.cash in safari Mobile web.
2. Long press any message to open the context menu.
3. You should not see any blue screen. (See the linked issue for example).

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Mobile Web

https://user-images.githubusercontent.com/24370807/124889537-d09e9480-dff4-11eb-9560-edda29080a7f.mp4

### CODE REVIEW

File: src/pages/home/report/ReportActionItem.js [hunk 1]
```diff
@@ -269,6 +272,8 @@ class ReportActionItem extends Component {
             <>
                 <PressableWithSecondaryInteraction
                     ref={el => this.popoverAnchor = el}
+                    onPressIn={() => this.props.isSmallScreenWidth && ControlSelection.block()}
+                    onPressOut={() => ControlSelection.unblock()}
```
Comment (TomatoToaster) [inline src/pages/home/report/ReportActionItem.js:276]: NAB: Why not include `this.props.isSmallScreenWidth && ` here too?

**Review by TomatoToaster — DISMISSED** [review]

**Comment by TomatoToaster** [comment]
LGTM, just had one comment lmk what you think.

File: src/pages/home/report/ReportActionItem.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/home/report/ReportActionItem.js:276]: This Change will block text selection at the root. But we have a feature where we select text on the web and then open the context menu to copy the selected part.

**Comment by parasharrajat** [comment]
FYI, I tried many ways to fix this issue. There is no simple way. We could have blocked the selection on the Lower message component. But that does not work. We already have that. In IOS, I have found that If the main target has selection blocked and you try to open the menu over it. Safari will expand your touch until it found something to select.

For this blue screen. There are two things that were messing around.
1. First the above-mentioned thing, Will cause the chat user name to be selected who is listed over the messages.
2. Secondly, when modal is shown, it overlays a background thin layer. But that layer will be lied underneath your touch so Safari will move that selection to that window. Which causes everything to be selected on the page.

So I have to use this solution as a hack to fix it. Which Keep other platforms intact.

**Comment by parasharrajat** [comment]
@TomatoToaster I would like to request you to test one thing before we merge it. When you open the context menu, try to close it by tapping on the background grayed area. It should close. Let me know if that has no issues. I just want to be sure that it does not break anything. I have an IOS simulator but that is not good to really test these things.

Please test it on Android, IOS, safari M-web. Thank you.

**Comment by parasharrajat** [comment]
@TomatoToaster Needs a push. Thanks.

**Comment by parasharrajat** [comment]
Looking forward to a merge if it looks good.

**Comment by TomatoToaster** [comment]
This still looks good, but this got some merge conflicts for you to resolve.

I tried testing this on my iOS device locally but it's not set up with our developer account and it turns out I need someone else to approve it 😞 . 

Instead I tested the modal closing on iOS and android through simulators. It looks like it's working fine and on safari's mweb too! For proper physical device testing, I think it will be fine to verify when it hits staging and we can actually download it.

**Comment by parasharrajat** [comment]
Ready. Updated

**Review by TomatoToaster — APPROVED** [review]

**Pull request merged**